lib/credentials/shared_ini_file_credentials.js

```diff
@@ -87,11 +87,8 @@ AWS.SharedIniFileCredentials = AWS.util.inherit(AWS.Credentials, {
         filename: this.filename ||
           (process.env[AWS.util.configOptInEnv] && process.env[AWS.util.sharedCredentialsFileEnv])
       });
-      for (var i = 0, availableProfiles = creds.getProfiles(); i < availableProfiles.length; i++) {
-        profiles[availableProfiles[i]] = AWS.util.merge(
-          profiles[availableProfiles[i]] || {},
-          creds.getProfile(availableProfiles[i])
-        );
+      for (i = 0, availableProfiles = creds.getProfiles(); i < availableProfiles.length; i++) {
```

Comment:
Nit: Can we either use `var i` here, or declare `var i` at the top of the function? Just worried if we someday change `i` in the for loop above, then this `i` becomes a global.